Origin: vendor
Subject: drop rust-rsa dependency because #1057096 / CVE-2023-49092
Forwarded: not-needed

this is an upstream ack'ed temporary solution while upstream is working on
a proper fix...

diff --git a/Cargo.toml b/Cargo.toml
index 902ada9..b812947 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -50,8 +50,8 @@ version = "0.5"
 [dependencies.openpgp-cert-d]
 version = "0.3.1"
 
-[dependencies.rsa]
-version = "0.9"
+#[dependencies.rsa]
+#version = "0.9"
 
 [dependencies.sequoia-keystore-backend]
 version = "0.6"
diff --git a/src/privkey.rs b/src/privkey.rs
index dcdfe0f..99100f9 100644
--- a/src/privkey.rs
+++ b/src/privkey.rs
@@ -19,8 +19,6 @@ use sequoia_openpgp::packet::{
 };
 use sequoia_openpgp::types::{Curve, Timestamp};
 
-use rsa::traits::PrivateKeyParts;
-
 /// A SequoiaKey represents the private cryptographic key material of an
 /// OpenPGP (sub)key to be uploaded to an OpenPGP card.
 pub(crate) struct SequoiaKey {
@@ -72,11 +70,13 @@ impl CardUploadableKey for SequoiaKey {
         let secret_key_material = unenc.map(|mpis| mpis.clone());
 
         match (self.public.clone(), secret_key_material) {
+            /*
             (mpi::PublicKey::RSA { e, n }, mpi::SecretKeyMaterial::RSA { d, p, q, u: _ }) => {
                 let sq_rsa = SqRSA::new(e, d, n, p, q)?;
 
                 Ok(PrivateKeyMaterial::R(Box::new(sq_rsa)))
             }
+            */
             (mpi::PublicKey::ECDH { curve, q, .. }, mpi::SecretKeyMaterial::ECDH { scalar }) => {
                 let sq_ecc = SqEccKey::new(curve, scalar, q, EccType::ECDH);
 
@@ -114,13 +114,13 @@ impl CardUploadableKey for SequoiaKey {
     }
 }
 
-fn mpi_to_biguint(mpi: &MPI) -> rsa::BigUint {
-    slice_to_biguint(mpi.value())
-}
+//fn mpi_to_biguint(mpi: &MPI) -> rsa::BigUint {
+//    slice_to_biguint(mpi.value())
+//}
 
-fn slice_to_biguint(bytes: &[u8]) -> rsa::BigUint {
-    rsa::BigUint::from_bytes_be(bytes)
-}
+//fn slice_to_biguint(bytes: &[u8]) -> rsa::BigUint {
+//    rsa::BigUint::from_bytes_be(bytes)
+//}
 
 /// RSA-specific data-structure to hold private (sub)key material for upload
 /// with the `openpgp-card` crate.
@@ -134,6 +134,7 @@ struct SqRSA {
     dq1: ProtectedMPI,
 }
 
+/*
 impl SqRSA {
     #[allow(clippy::many_single_char_names)]
     fn new(
@@ -182,6 +183,7 @@ impl SqRSA {
         })
     }
 }
+*/
 
 impl RSAKey for SqRSA {
     fn e(&self) -> &[u8] {
